src/account/BiconomySmartAccountV2.ts

```diff
@@ -1372,26 +1371,46 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
   /**
    * Transfers ownership of the smart account to a new owner.
    * @param newOwner The address of the new owner.
+   * @param moduleAddress The address of the validation module (ECDSA Ownership Module or Multichain Validation Module).
    * @param buildUseropDto {@link BuildUserOpOptions}. Optional parameter
-   * @returns A Promise that resolves to a TransferOwnershipResponse or rejects with an Error.
+   * @returns A Promise that resolves to a UserOpResponse or rejects with an Error.
+   * @description This function will transfer ownership of the smart account to a new owner. If you use session key manager module, after transferring the ownership
+   * you will need to re-create a session for the smart account with the new owner (signer) and specify "accountAddress" in "createSmartAccountClient" function.
    * @example
    * ```typescript
-   * const walletClient = createWalletClient({
-   *   account,
-   *   chain: baseSepolia,
-   *   transport: http()
-   * });
-   * const smartAccount = await createSmartAccountClient({
-   *   signer: walletClient,
-   *   paymasterUrl: "https://paymaster.biconomy.io/api/v1/...",
-   *   bundlerUrl: `https://bundler.biconomy.io/api/v2/84532/nJPK7B3ru.dd7f7861-190d-41bd-af80-6877f74b8f44`,
-   *   chainId: 84532
-   * });
-   * const response = await smartAccount.transferOwnership(newOwner, {paymasterServiceData: {mode: PaymasterMode.SPONSORED}});
+   * 
+   * let walletClient = createWalletClient({
+        account,
+        chain: baseSepolia,
+        transport: http()
+      });
+
+      let smartAccount = await createSmartAccountClient({
+        signer: walletClient,
+        paymasterUrl: "https://paymaster.biconomy.io/api/v1/...",
+        bundlerUrl: `https://bundler.biconomy.io/api/v2/84532/nJPK7B3ru.dd7f7861-190d-41bd-af80-6877f74b8f44`,
+        chainId: 84532
+      });
+      const response = await smartAccount.transferOwnership(newOwner, DEFAULT_ECDSA_OWNERSHIP_MODULE, {paymasterServiceData: {mode: PaymasterMode.SPONSORED}});
+      
+      walletClient = createWalletClient({
+        newOwnerAccount,
+        chain: baseSepolia,
+        transport: http()
+      })
+      
+      smartAccount = await createSmartAccountClient({
+        signer: walletClient,
+        paymasterUrl: "https://paymaster.biconomy.io/api/v1/...",
+        bundlerUrl: `https://bundler.biconomy.io/api/v2/84532/nJPK7B3ru.dd7f7861-190d-41bd-af80-6877f74b8f44`,
+        chainId: 84532,
+        accountAddress: await smartAccount.getAccountAddress()
+      })
    * ```
    */
   async transferOwnership(
```

Comment:
since you want to modularize it,
I'd say instead of newOwner make it an object called newOwnershipInfo and have eoa pubkey (newOwner) as one of possible args 
encodedCall call also be just passed optional which will directly call the chosen module. currently encoded call only builds for transferOwnership(address _newower) which is only applicable for moduleAddress passed as ECDSA or Multi-chain 

or we can just let it be as this is. food for thought :)

Reply:
The usecase is only for ECDSA and Multi Chain anyway, I would prefer to let it as it is. If we allow encodedCall as an argument, then anything can be passed to it, this method should only allow to do what it says, a transfer of ownership. Lmk what you think @livingrockrises

Reply:
yes we can let it be and be precise on the description that module address passed would be ecdsa or multi-chain (any module which has single eoa owner and has transferOwnership method in it)

Reply:
Added a type for this argument, to allow only ECDSA and Multichain module address